src/Generator/Traits/ChangelogTemplate.php

```diff
@@ -0,0 +1,174 @@
+<?php
+namespace Mill\Generator\Traits;
+
+use StringTemplate\Engine;
+
+trait ChangelogTemplate
+{
+    /**
+     * Changelog template output format. Acceptable values: "json", "html", or "markdown"
+     *
+     * @var string
+     */
+    protected $output_format = 'json';
+
+    /**
+     * @var Engine|null
+     */
+    protected $template_engine;
+
+    /**
+     * Render a template with some content.
+     *
+     * @param string $template
+     * @param array $content
+     * @return string
+     */
+    protected function renderText($template, array $content = [])
+    {
+        if (is_null($this->template_engine)) {
+            $this->template_engine = new Engine;
+        }
+
+        if (in_array($this->output_format, ['json', 'html'])) {
+            list($template, $content) = $this->transformTemplateIntoHtml($template, $content);
+        } else {
+            list($template, $content) = $this->transformTemplateIntoMarkdown($template, $content);
+        }
+
+        return $this->template_engine->render($template, $content);
+    }
+
+    /**
+     * Transform a template by wrapping specific content in styleable HTML elements.
+     *
+     * @param string $template
+     * @param array $content
+     * @return array
+     */
+    protected function transformTemplateIntoHtml($template, array $content = [])
+    {
+        $searches = [];
+        $replacements = [];
+        foreach ($content as $key => $value) {
+            $data_attribute_key = str_replace('_', '-', $key);
+
+            switch ($key) {
+                case 'content_type':
+                case 'field':
+                case 'group':
+                case 'http_code':
+                case 'method':
+                case 'parameter':
+                case 'representation':
+                case 'uri':
+                    $searches[] = '{' . $key . '}';
+                    if (is_array($value)) {
+                        $replacements[] = $this->joinWords(
+                            array_map(function ($val) use ($data_attribute_key, $key) {
```

Comment:
maybe pull this out into a descriptively named function for legibility and to improve reading comprehension?